app/models/manager_refresh/inventory/target.rb

```diff
@@ -1,14 +1,70 @@
 class ManagerRefresh::Inventory::Target
-  # @param [ApplicationRecord] root
-  def initialize(root)
-    @root = root
-  end
+  attr_reader :collector, :collections
+
+  delegate :manager, :target, :to => :collector
 
-  def collections
-    @collections ||= {}
+  # @param collector [ManagerRefresh::Inventory::Collector] A collector object
+  def initialize(collector)
```

Comment:
not sure if I like the collector being tied to this class.
This increases the coupling